app/src/main/java/com/example/androidshowcase/ui/insidelibrary/ComponentsRecyclerAdapter.kt

```diff
@@ -4,15 +4,21 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.CheckBox
+import android.widget.CompoundButton
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
 import com.example.androidshowcase.R
 import com.example.androidshowcase.database.entities.Component
+import com.example.androidshowcase.database.entities.ComponentMarking
+import com.example.androidshowcase.database.entities.MarkingType
 
-class ComponentsRecyclerAdapter : RecyclerView.Adapter<ComponentsRecyclerAdapter.ViewHolder>() {
+class ComponentsRecyclerAdapter(argMarkings: List<ComponentMarking>, argMarkingTypes: List<MarkingType>) : RecyclerView.Adapter<ComponentsRecyclerAdapter.ViewHolder>() {
```

Comment:
Don't pass data in constructor, create fun setData(argMarkings: List<ComponentMarking>, argMarkingTypes: List<MarkingType>)